extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/AuthenticationFailedExceptionMapper.java

```diff
@@ -37,18 +42,25 @@ public Response toResponse(AuthenticationFailedException exception) {
             if (authenticator != null) {
                 ChallengeData challengeData = authenticator.getChallenge(context)
                         .await().indefinitely();
-                Response.ResponseBuilder status = Response.status(challengeData.status);
-                if (challengeData.headerName != null) {
+                int statusCode = challengeData == null ? 401 : challengeData.status;
+                Response.ResponseBuilder status = Response.status(statusCode);
```

Comment:
This is all right, and I see it was here before, but I've found it a bit confusing that the response builder is named `status` :-), suggest to rename it to `response`

Reply:
I used `responseBuilder` instead. IMHO it looks better than `status`, so thank you for the suggestion.